### PR TITLE
Fix haddock comments and bump hspec upper bound

### PR DESCRIPTION
**Pull request by cocreature**

### CODE REVIEW

File: Data/GraphViz/Types/Graph.hs [hunk 1]
```diff
@@ -231,9 +231,9 @@ emptyGA = GA S.empty S.empty S.empty
     dg' = addNode n mc as dg
 
     merge = addSuccRev n ps'' . addPredRev n ss''
-            -- ^ Add reverse edges
```
Comment (ivan-m) [inline Data/GraphViz/Types/Graph.hs]: Whilst these aren't processed by Haddock, I prefer to keep the carets here to point to the line the comment is referring to (making it a bit more obvious what I'm talking about).

**Comment by ivan-m** [comment]
Unless Haddock actually complains about those comments, I'd prefer to keep them as-is for readability purposes.

**Comment by cocreature** [comment]
Sorry I should have made this more clear: These comments break haddock and are the reason for why there are currently haddocks on hackage.

**Comment by ivan-m** [comment]
Is this something new in Haddock (as it hasn't been an issue before)? Sounds like a bug to me.

**Comment by cocreature** [comment]
No idea, sry.

**Comment by ivan-m** [comment]
This seems related to haskell/haddock#144 and similar issues; i.e. it's GHC's fault.

**Pull request merged**